### ISSUE TITLE
💡 Idea: Add controller preference to ignore FX / Master tracks in session?

### ISSUE THREAD

**Issue by Fannon** [weskoop/Launchpad-Pro-Mk3-for-Bitwig#6]
Just an idea to throw in: There could be controller preferences whether you want to ignore the Effect Tracks or the Master track in session view.

**Comment by weskoop** [comment]
Adding more features and options, means more deviations from the defined usage by Novation, and starts to get away from the goals of this project in the first place.

That said, I'd be open to this an an option if you want to PR, but it may over-complicate things if you still want access to Volume, panning, etc. on those other tracks, then we may need multiple TrackBanks, and we'll need to change the target of the arrows, etc. If it starts to get hairy, I'd probably avoid it.

**Comment by Fannon** [comment]
Yes, I get your point and agree that this shouldn't get too complicated. Otherwise it would not be a good fit for the goal of staying simple.

I assumed that if you ignore those tracks, you'll also loose access to them entirely. If this is OK or not is up to you.
If you only want to play session mode, it might be OK and actually very similar to what you do with https://intro.novationmusic.com/

**Comment by weskoop** [comment]
@Fannon I would take a PR, if it fully swaps the trackBank across the board. I don't want follow and manage separate banks for the clip launcher and faders/devices.

That means you would lose access to select tracks/devices on send tracks and the Master, most people will have only 1 or 2 sends and there is only one master track, so for many, that's just 3 tracks at the end, and you have fader control over all of them, so i find it useful to have all the tracks. I also think if you are the type to have 5 sends, you may want some control over those.

If you still think that's still useful for someone let me know, otherwise let's close this.

**Comment by Fannon** [comment]
Ok, let's close this. It sounds like it would be more complicated to implement.

If I find that I really need it, I could try it and create a PR.
Right now, I'm only using it at home for DAW work and in this setting the current mode is working fine.

**Comment by weskoop** [comment]
Been thinking a lot about this, especially as I try and fix the sends layer.

I went back into Ableton to see how it worked there and the LaunchPad does indeed ignore the effects and master tracks. This will both simplify the code (less special cases for the Sends Page) and stick closer to the original design.

My original reply may have been a bit hasty, but I still think it should be one way or the other, not a preference

TL;DR — I think Ignoring the Effects/Master tracks makes sense, no preference needed.

**Comment by weskoop** [comment]
Wow, this inadvertently fixed another bug I've been fighting with, this feels like the right move for sure and I can toss the spaghetti code I was pushing against on Saturday.
